app/classes/knight.py

```diff
@@ -0,0 +1,32 @@
+class Knight:
+    def __init__(self, knight: dict) -> None:
+        self.name = knight["name"]
+        self.power = knight["power"]
+        self.hp = knight["hp"]
+        self.armour = knight["armour"]
+        self.weapon = knight["weapon"]
+        self.potion = knight["potion"]
+        self.protection = 0
+
+    def apply_weapon(self) -> None:
+        self.power += self.weapon["power"]
+
+    def apply_armour(self) -> None:
+        for armo in self.armour:
+            self.protection += armo["protection"]
+
+    def apply_potion_if_exist(self) -> None:
```

Comment:
There is no need to include implementation details in the function name
```suggestion
    def apply_potion(self) -> None:
```